core/deployment/src/main/java/io/quarkus/deployment/dev/testing/TestRunner.java

```diff
@@ -0,0 +1,348 @@
+     * <p>
+     * We manually check for application.properties changes and apply them.
+     */
+    private void handleApplicationPropertiesChange() {
+        for (Path rootPath : testApplication.getQuarkusBootstrap().getApplicationRoot()) {
+            Path appProps = rootPath.resolve("application.properties");
+            if (Files.exists(appProps)) {
+                Properties p = new Properties();
+                try (InputStream in = Files.newInputStream(appProps)) {
+                    p.load(in);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+                String includeTags = p.getProperty("quarkus.test.include-tags");
+                String excludeTags = p.getProperty("quarkus.test.exclude-tags");
+                String includePattern = p.getProperty("quarkus.test.include-pattern");
+                String excludePattern = p.getProperty("quarkus.test.exclude-pattern");
+                if (!firstRun) {
+                    if (!Objects.equals(includeTags, appPropertiesIncludeTags)) {
+                        if (includeTags == null) {
+                            testSupport.includeTags = Collections.emptyList();
+                        } else {
+                            testSupport.includeTags = Arrays.stream(includeTags.split(",")).map(String::trim)
+                                    .collect(Collectors.toList());
+                        }
+                    }
+                    if (!Objects.equals(excludeTags, appPropertiesExcludeTags)) {
+                        if (excludeTags == null) {
+                            testSupport.excludeTags = Collections.emptyList();
+                        } else {
+                            testSupport.excludeTags = Arrays.stream(excludeTags.split(",")).map(String::trim)
+                                    .collect(Collectors.toList());
+                        }
+                    }
+                    if (!Objects.equals(includePattern, appPropertiesIncludePattern)) {
+                        if (includePattern == null) {
+                            testSupport.include = null;
+                        } else {
+                            testSupport.include = Pattern.compile(includePattern);
+                        }
+                    }
+                    if (!Objects.equals(excludePattern, appPropertiesExcludePattern)) {
+                        if (excludePattern == null) {
+                            testSupport.exclude = null;
+                        } else {
+                            testSupport.exclude = Pattern.compile(excludePattern);
+                        }
+                    }
+                }
+                appPropertiesIncludeTags = includeTags;
+                appPropertiesExcludeTags = excludeTags;
+                appPropertiesIncludePattern = includePattern;
+                appPropertiesExcludePattern = excludePattern;
+            }
```

Comment:
A minor one, you could break the loop once the `application.properties` has been found.